bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/handler/VehicleHandler.java

```diff
@@ -198,336 +165,733 @@ public void initialize() {
-            return;
-        }
-        String token = accountHandler.get().getToken();
-        if (token.isEmpty()) {
-            String textKey = Constants.STATUS_TEXT_PREFIX + "vehicle" + Constants.STATUS_BRIDGE_ATHORIZATION;
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, textKey);
-            return;
-        } else if (!online) { // only update if thing isn't already ONLINE
-            updateStatus(ThingStatus.ONLINE);
-        }
-
-        // Mileage for all cars
-        String odoUrl = String.format(ODO_URL, config.get().vin);
-        if (accountConfigAvailable()) {
-            if (accountHandler.get().config.get().odoScope) {
-                call(odoUrl);
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        /**
+         * Commands shall be not that frequent so trace level for identifying problems should be feasible
+         */
+        logger.trace("Received command {} {} for {}", command.getClass(), command, channelUID);
+        if (command instanceof RefreshType) {
+            if (MB_KEY_FEATURE_CAPABILITIES.equals(channelUID.getIdWithoutGroup())
+                    || MB_KEY_COMMAND_CAPABILITIES.equals(channelUID.getIdWithoutGroup())) {
+                accountHandler.ifPresent(ah -> {
+                    ah.getVehicleCapabilities(config.get().vin);
+                });
             } else {
-                logger.trace("{} Odo scope not activated", this.getThing().getLabel());
+                // deliver from event storage
+                ChannelStateMap csm = eventStorage.get(channelUID.getId());
+                if (csm != null) {
+                    updateChannel(csm);
+                }
             }
-        } else {
-            logger.trace("{} Account not properly configured", this.getThing().getLabel());
-        }
-
-        // Electric status for hybrid and electric
-        if (uid.equals(BEV) || uid.equals(HYBRID)) {
-            String evUrl = String.format(EV_URL, config.get().vin);
-            if (accountConfigAvailable()) {
-                if (accountHandler.get().config.get().evScope) {
-                    call(evUrl);
+            // ensure unit update
+            unitStorage.remove(channelUID.getIdWithoutGroup());
+        } else if (Constants.GROUP_VEHICLE.equals(channelUID.getGroupId())) {
+            /**
+             * Commands for Vehicle
+             */
+            if (OH_CHANNEL_IGNITION.equals(channelUID.getIdWithoutGroup())) {
+                String supported = thing.getProperties().get(MB_KEY_COMMAND_ENGINE_START);
+                if (Boolean.FALSE.toString().equals(supported)) {
+                    logger.trace("Engine Start/Stop not supported");
                 } else {
-                    logger.trace("{} Electric Status scope not activated", this.getThing().getLabel());
+                    int commandValue = ((DecimalType) command).intValue();
```

Comment:
This is the reason why the handler is such a big class. `handleCommand` is nearly half of the code and contains much of dumb code
- is command supported
- is PIN required and set
- each command needs a dedicated proto message

I didn't took care about casting because the channel is declared as Number so it should be a `DecimalType`. I shortly tested with a String and got the result below. 
Invalid commands seems to be blocked on higher level. `RefreshType` is covered on top of `handleCommand `and  then casting should be fine according to channel definition.

```
23:37:26.639 [WARN ] [ab.core.model.script.actions.BusEvent] - Cannot convert 'WRONG_COMMAND' to a command type which item 'Mercedes_Benz_EQA_Charge_Program' accepts: [DecimalType, QuantityType, RefreshType].
```